### ISSUE TITLE
[Feature] Introduces savepoint mechanism of Paimon

### ISSUE THREAD

**Issue by JingsongLi** [apache/incubator-paimon#748]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

Disaster Recovery is very much mission critical for any software. Especially when it comes to data systems, the impact could be very serious leading to delay in business decisions or even wrong business decisions at times. Paimon could introduce savepoint mechanism to assist users in recovering data from a previous state.

As the name suggest, "savepoint" saves the table as of the snapshot, so that it lets you restore the table to this savepoint at a later point in snapshot if need be. Care is taken to ensure cleaner will not clean up any files that are savepointed. On similar lines, savepoint cannot be triggered on a snapshot that is already cleaned up. In simpler terms, this is synonymous to taking a backup, just that we don't make a new copy of the table, but just save the state of the table elegantly so that we can restore it later when in need.

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by schnappi17** [comment]
@JingsongLi Please assign this task to me, thank you~

**Comment by JingsongLi** [comment]
Thanks @schnappi17 , this may need a PIP. https://cwiki.apache.org/confluence/display/PAIMON/Paimon+Improvement+Proposals

**Comment by schnappi17** [comment]
Got it. @JingsongLi

**Comment by FangYongs** [comment]
Use tag instead of savepoint
